torch/distributed/_composable/contract.py

```diff
@@ -149,4 +174,10 @@ def get_state(module: nn.Module) -> Optional[_State]:
 
     wrapper.state = get_state  # type: ignore[attr-defined]
 
+    def get_registry(module: nn.Module) -> OrderedDict[str, RegistryItem]:
+        default_registry: Dict[str, RegistryItem] = OrderedDict()
+        return module.__dict__.setdefault(REGISTRY_KEY, default_registry)
+
+    wrapper.registry = get_registry
```

Comment:
In this way, when we try to get registry, we do `replicate.registry(m)` or `fully_shard.registry(m)`, but they return the same dict. Does it make sense to move this function to a top level API, say `contract.registry(m)`?

Reply:
This is a good point. I do not feel strongly that we must make it `contract.registry(m)`, but I agree with @yhcharles that that API would be cleaner.